test/h-group/level-11.js

```diff
@@ -0,0 +1,299 @@
+		takeTurn(game, 'Donald plays g1 (slot 2)', 'r1');
+		takeTurn(game, 'Alice plays b1 (slot 4)', 'r3');
+		takeTurn(game, 'Bob clues green to Donald (slot 4)');
+
+		// Cathy's slot 1 could be any playable.
+		assert.equal(game.common.thoughts[game.state.hands[PLAYER.CATHY][0].order].finessed, true);
+		ExAsserts.cardHasInferences(game.common.thoughts[game.state.hands[PLAYER.CATHY][0].order], ['r1', 'y1', 'g2', 'b2', 'p1']);
+		assert.equal(game.common.thoughts[game.state.hands[PLAYER.ALICE][0].order].finessed, false);
+
+		// Donald's slot 4 must be g2,g3.
+		ExAsserts.cardHasInferences(game.common.thoughts[game.state.hands[PLAYER.DONALD][3].order], ['g2', 'g3']);
+
+		takeTurn(game, 'Cathy plays p1 (slot 1)', 'p5');
+
+		// After Cathy plays into the bluff, Donald knows it is a g3.
+		ExAsserts.cardHasInferences(game.common.thoughts[game.state.hands[PLAYER.DONALD][3].order], ['g3']);
+
+		// And no-one is finessed.
+		assert.equal(game.common.thoughts[game.state.hands[PLAYER.ALICE][0].order].finessed, false);
+		assert.equal(game.common.thoughts[game.state.hands[PLAYER.BOB][0].order].finessed, false);
+		assert.equal(game.common.thoughts[game.state.hands[PLAYER.CATHY][0].order].finessed, false);
+		assert.equal(game.common.thoughts[game.state.hands[PLAYER.DONALD][0].order].finessed, false);
+	});
+
+	it('understands receiving a bluff', () => {
+		const game = setup(HGroup, [
+			['xx', 'xx', 'xx', 'xx', 'xx'],
+			['b4', 'r1', 'y1', 'g5', 'p2'],
+			['p1', 'r4', 'b5', 'b2', 'y4']
+		], {
+			level: 11,
+			play_stacks: [2, 2, 2, 2, 2],
+			starting: PLAYER.CATHY
+		});
+		takeTurn(game, 'Cathy clues blue to Bob (slot 1)');
+
+		// Alice's slot 1 could be any of the playable 3's.
+		assert.equal(game.common.thoughts[game.state.hands[PLAYER.ALICE][0].order].finessed, true);
+		ExAsserts.cardHasInferences(game.common.thoughts[game.state.hands[PLAYER.ALICE][0].order], ['r3', 'y3', 'g3', 'b3', 'p3']);
+		// Bob's slot 1 must be b4.
+		ExAsserts.cardHasInferences(game.common.thoughts[game.state.hands[PLAYER.BOB][0].order], ['b4']);
+
+		takeTurn(game, 'Alice plays b3 (slot 1)', 'y5');
+
+		// After Alice plays into the bluff, Bob knows it is a b4 
+		ExAsserts.cardHasInferences(game.common.thoughts[game.state.hands[PLAYER.BOB][0].order], ['b4']);
+	});
+
+	it('never assumes a bluff when reverse finesse exists', () => {
+		const game = setup(HGroup, [
+			['xx', 'xx', 'xx', 'xx', 'xx'],
+			['b4', 'r1', 'y1', 'g5', 'p2'],
+			['b3', 'w1', 'w1', 'w5', 'w2'],
+			['p1', 'r4', 'b5', 'b2', 'y4']
+		], {
+			level: 11,
+			play_stacks: [2, 2, 2, 2, 2],
+			starting: PLAYER.DONALD
+		});
+		takeTurn(game, 'Donald clues blue to Bob (slot 1)');
+
+		// Alice's slot 1 should is not assumed
+		assert.equal(game.common.thoughts[game.state.hands[PLAYER.ALICE][0].order].finessed, false);
+		// Bob will know this isb4.
+		ExAsserts.cardHasInferences(game.common.thoughts[game.state.hands[PLAYER.BOB][0].order], ['b3', 'b4']);
```

Comment:
The comment here says that Bob should know b4, but I think the code is correct in only assuming [b3,b4] on Bob's slot 1.

Reply:
Updated the comments, by "will know" I meant after the play happens. Of course if the play doesn't happen Bob will know that it's b3.